src/Swarm/Language/Pretty.hs

```diff
@@ -188,7 +188,9 @@ instance PrettyPrec Term where
   prettyPrec _ (TVar s) = pretty s
   prettyPrec _ (TDelay _ t) = group . encloseWithIndent 2 lbrace rbrace $ ppr t
   prettyPrec _ t@TPair {} = prettyTuple t
-  prettyPrec _ t@(TLam {}) = prettyLambdas t
+  prettyPrec p t@(TLam {}) =
+    pparens (p > 9) $
+      prettyLambdas t
```

Comment:
@byorgey I assume this is overly pessimistic - in the example case, only the inner lambda requires parentheses.

However, the least amount of parentheses might be harder to read. 😄

Reply:
Hah, yes, in theory, a lambda at the very tail end of an expression does not need parentheses, but that would be annoying to read.  I think this is fine.